### DISCUSSION TITLE
Error: TypeError: Cannot read property 'split' of undefined when using Tailwind CSS with WordPress theme

### DISCUSSION THREAD

**Discussion by imadhnini** [tailwindlabs/tailwindcss#14325] [Help]
Hi everyone,

I'm encountering an issue while trying to start Tailwind CSS in my WordPress theme. When I run the following command:

`npx tailwindcss -i ./assets/css/tailwind-config.css -o ./assets/css/tailwind.css --watch`

I get the following error:

```
TypeError: Cannot read property 'split' of undefined
    at getClassCandidates (C:\Users\finoi\AppData\Roaming\npm-cache\_npx\9456\node_modules\tailwindcss\lib\lib\expandTailwindAtRules.js:88:30)
    at C:\Users\finoi\AppData\Roaming\npm-cache\_npx\9456\node_modules\tailwindcss\lib\lib\expandTailwindAtRules.js:174:17
    at Array.map (<anonymous>)
    at C:\Users\finoi\AppData\Roaming\npm-cache\_npx\9456\node_modules\tailwindcss\lib\lib\expandTailwindAtRules.js:172:37
    at C:\Users\finoi\AppData\Roaming\npm-cache\_npx\9456\node_modules\tailwindcss\lib\lib\processTailwindFeatures.js:51:59
    at Object.Once (C:\Users\finoi\AppData\Roaming\npm-cache\_npx\9456\node_modules\tailwindcss\lib\cli\build\plugin.js:257:19)
    at LazyResult.runOnRoot (C:\Users\finoi\AppData\Roaming\npm-cache\_npx\9456\node_modules\tailwindcss\node_modules\postcss\lib\lazy-result.js:327:23)
    at LazyResult.runAsync (C:\Users\finoi\AppData\Roaming\npm-cache\_npx\9456\node_modules\tailwindcss\node_modules\postcss\lib\lazy-result.js:258:26)
    at async Object.watch (C:\Users\finoi\AppData\Roaming\npm-cache\_npx\9456\node_modules\tailwindcss\lib\cli\build\plugin.js:370:13)
    at async build (C:\Users\finoi\AppData\Roaming\npm-cache\_npx\9456\node_modules\tailwindcss\lib\cli\build\index.js:47:9)
```

Has anyone encountered this error before or have any suggestions on how to resolve it? Any help would be greatly appreciated!

Thanks in advance!

**Comment by wongjn** [comment]
Would you be able to provide a project that reproduces the unexpected behavior please? Or at least the content of `./assets/css/tailwind-config.css` and your Tailwind configuration file.

**Reply by imadhnini** [reply]
content of `./assets/css/tailwind-config.css` file

```
@tailwind base;
@tailwind components;
@tailwind utilities;
```

content of tailwind-config.js file

```
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    ["./*.php", "./woocommerce/*.php", "./parts/*.php", "./admin/parts/*.php"],
  ],
  theme: {
    extend: {},
  },
  plugins: [],
};
```

**Reply by wongjn** [reply]
Thank you. So several things:
- Your `tailwind-config.js` should be named `tailwind.config.js`
- Your `content` is malformed. It should only be a single-depth array:
  ```diff
   module.exports = {
     content: [
  -    ["./*.php", "./woocommerce/*.php", "./parts/*.php", "./admin/parts/*.php"],
  +    "./*.php",
  +    "./woocommerce/*.php",
  +    "./parts/*.php",
  +    "./admin/parts/*.php",
     ],
     theme: {
  ```
  
It could also possibly be a malformed class name in one of the files scanned by Tailwind. Consider checking you are not using any sort of [string interpolation/concatenation to build any class names](https://tailwindcss.com/docs/content-configuration#dynamic-class-names). Otherwise, consider:
- using a process of elimination to identify the file(s) that may be causing the Tailwind error.
- providing a project that reproduces the unexpected behavior.

**Reply by imadhnini** [reply]
I did all what you said and still same problem. I don't know how to resolve it 😑.

**Reply by wongjn** [reply]
Consider providing a project that reproduces the unexpected behavior.
